### PR TITLE
feature: finish restart interface

### PR DESCRIPTION
**Pull request by HusterWan**
Signed-off-by: Michael Wan <zirenwan@gmail.com>

### Ⅰ. Describe what this PR did
finish restart interface daemon part

### CODE REVIEW

**Comment by HusterWan** [comment]
the is a problem: If start container too quickly after stop it, the snapshot of container will be deleted by the containerd gc.  this pr should not be merged utill i fix the problem

**Comment by HusterWan** [comment]
> there is a problem: If start container too quickly after stop it, the snapshot of the container will be deleted by the containerd gc. this pr should not be merged utill i fix the problem

i have figured it out:  we should add lease in ctx, when we stop a container to prevent the snapshot being deleted by gc.

File: daemon/mgr/container.go [hunk 1]
```diff
@@ -897,6 +908,34 @@ func (mgr *ContainerManager) Resize(ctx context.Context, name string, opts types
...
+
+	if timeout == 0 {
+		timeout = c.StopTimeout()
+	}
+
+	// stop container
+	if err := mgr.stop(ctx, c, timeout); err != nil {
+		return errors.Wrapf(err, "failed to stop container")
+	}
+	logrus.Debug("Restart: container " + c.ID() + "  stopped succeeded")
```
Comment (allencloud) [inline daemon/mgr/container.go]: we usually use `logrus.Debugf()` here.

**Comment by allencloud** [comment]
LGTM

**Pull request merged**